x/compliance/handler_test.go

```diff
@@ -1240,6 +1240,77 @@ func TestHandler_CertifyRevokedModel(t *testing.T) {
 	}
 }
 
+func TestHandler_DeleteComplianceInfoForAllCertStatuses(t *testing.T) {
+	setup := Setup(t)
+
+	// add model version
+	vid, pid, softwareVersion, softwareVersionString := setup.AddModelVersion(
+		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion, testconstants.SoftwareVersionString)
+
+	// certify model
+	certifyModelMsg := NewMsgCertifyModel(
+		vid, pid, softwareVersion, softwareVersionString, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err := setup.Handler(setup.Ctx, certifyModelMsg)
+	require.NoError(t, err)
+
+	deleteComplInfoMsg := NewMsgDeleteComplianceInfo(vid, pid, softwareVersion, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, deleteComplInfoMsg)
+	require.NoError(t, err)
+
+	// check
+	certifiedModel, err := queryCertifiedModel(setup, vid, pid, softwareVersion, dclcompltypes.MatterCertificationType)
+	require.Equal(setup.T, err.Error(), "rpc error: code = NotFound desc = not found")
+	require.Nil(setup.T, certifiedModel)
+
+	// add model version
+	vid, pid, softwareVersion, softwareVersionString = setup.AddModelVersion(
+		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion+1, testconstants.SoftwareVersionString)
+
+	// Provision model
+	provisionModelMsg := NewMsgProvisionModel(
+		vid, pid, softwareVersion, softwareVersionString, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, provisionModelMsg)
+	require.NoError(t, err)
+
+	deleteComplInfoMsg = NewMsgDeleteComplianceInfo(vid, pid, softwareVersion, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, deleteComplInfoMsg)
+	require.NoError(t, err)
+
+	provisionalModel, err := queryProvisionalModel(setup, vid, pid, softwareVersion, dclcompltypes.MatterCertificationType)
+	require.Equal(setup.T, err.Error(), "rpc error: code = NotFound desc = not found")
+	require.Nil(setup.T, provisionalModel)
+
+	// add model version
+	vid, pid, softwareVersion, softwareVersionString = setup.AddModelVersion(
+		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion+2, testconstants.SoftwareVersionString)
+
+	// Revoke model
+	revokeModelMsg := NewMsgRevokeModel(
+		vid, pid, softwareVersion, softwareVersionString, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, revokeModelMsg)
+	require.NoError(t, err)
+
+	deleteComplInfoMsg = NewMsgDeleteComplianceInfo(vid, pid, softwareVersion, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, deleteComplInfoMsg)
+	require.NoError(t, err)
+
+	revokedModel, err := queryRevokedModel(setup, vid, pid, softwareVersion, dclcompltypes.MatterCertificationType)
+	require.Equal(setup.T, err.Error(), "rpc error: code = NotFound desc = not found")
+	require.Nil(setup.T, revokedModel)
```

Comment:
Please also check in all three cases (certified, provision, revoked), that
1) query by CDCertificateID returns Not Found
2) query the ComplianceInfo returns Not Found

Reply:
Also it would be better to have separate tests for every status